sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClientDiagnostics.cs

```diff
@@ -33,29 +33,28 @@ public TextAnalyticsClientDiagnostics(ClientOptions options) : base(options)
             ResponseHeaders responseHeaders,
             ref IDictionary<string, string>? additionalInfo)
         {
-            //if (!string.IsNullOrEmpty(content))
-            //{
-            //    try
-            //    {
-            //        // Try to parse the failure content and use that as the
-            //        // default value for the message, error code, etc.
-            //        using JsonDocument doc = JsonDocument.Parse(content);
-            //        if (doc.RootElement.TryGetProperty("error", out JsonElement errorElement))
-            //        {
-            //            TextAnalyticsError error = Transforms.ConvertToError(TextAnalyticsErrorInternal.DeserializeTextAnalyticsErrorInternal(errorElement));
+            if (!string.IsNullOrEmpty(content))
+            {
+                try
+                {
+                    // Try to parse the failure content and use that as the
+                    // default value for the message, error code, etc.
+                    using JsonDocument doc = JsonDocument.Parse(content);
+                    if (doc.RootElement.TryGetProperty("error", out JsonElement errorElement))
+                    {
+                        TextAnalyticsError error = Transforms.ConvertToError(Error.DeserializeError(errorElement));
```

Comment:
Good point. We could always overwrite the `DeserializeError` method. I avoided that in the past in case the Service kept modifying the structure of it. But this is not going to change too much, especially now that `Error` is a common type across services and follows the REST guidelines.

Issue https://github.com/Azure/azure-sdk-for-net/issues/28461